packages/react/src/checkbox-group/CheckboxGroup.tsx

```diff
@@ -75,11 +75,12 @@ export const CheckboxGroup = React.forwardRef(function CheckboxGroup(
   const id = useBaseUiId(idProp);
 
   const controlRef = React.useRef<HTMLButtonElement>(null);
-  const registerControlRef = useEventCallback((element: HTMLButtonElement | null) => {
+
+  const registerControlRef = React.useCallback((element: HTMLButtonElement | null) => {
     if (controlRef.current == null && element != null && !element.hasAttribute(PARENT_CHECKBOX)) {
       controlRef.current = element;
     }
-  });
+  }, []);
```

Comment:
Why is this changed? `useEventCallback` makes the intent more clear and is more performant.

Reply:
@michaldudak wrote that `useCallback` can be more performant

https://github.com/mui/base-ui/pull/2336#discussion_r2354908410

Reply:
It makes sense now that I re-read the source code, `useEventCallback` has to deal with init/render cycles. We could add something like `useUntrackedCallback` (name tbd) as a version of `React.useCallback` for functions without tracked dependencies like this one.